src/d3a/d3a_core/util.py

```diff
@@ -297,15 +297,16 @@ def update_nested_settings(class_object, class_name, settings_dict):
         update_nested_settings(setting_class, settings_class_name, advanced_settings)
 
 
-def generate_market_slot_list(area):
+def generate_market_slot_list(area=None):
     """
     Returns a list of all slot times
     """
+    config = GlobalConfig if area is None else area.config
```

Comment:
Can you help me understand, how could it help reduce the RAM requirement?

Reply:
This does not reduce the requirement of RAM. This change was only done in order to have a list of time_slots for the creation of the states (PV, Load, Storage), when there is not a `config` member in self.area yet.
In this case we use the GlobalConfig now, which is updated before the creation of the device instances.
Does this make sense?